app-near/src/server/typeform/handleTypeformAnswer.ts

```diff
@@ -21,39 +39,78 @@ export const handleTypeformAnswer = async (
       JSON.parse(body),
     );
     formId = parsedBody.form_response.form_id;
-    console.debug("[whebhook]", formId, body);
+    const typeformType = getFormIdType(formId);
+
+    console.debug("[whebhook]", typeformType, body);
 
     if (!isValidSignature(req, body, SignatureType.TYPEFORM)) {
-      return unauthorizedResponse(formId);
+      throw new TRPCError({
+        code: "UNAUTHORIZED",
+        message: ErrorCode.WRONG_SIGNATURE,
+      });
     }
 
-    const referencesMapping = getReferencesMapping(formId);
-    if (!referencesMapping) {
-      return referenceMappingNotFoundResponse(formId);
-    }
+    const referencesMapping = getReferencesMapping(typeformType);
 
     const answers = convertFormToAnswer(parsedBody, referencesMapping);
-    console.debug("[whebhook]", formId, JSON.stringify(answers));
-
-    if (formId === env.SU_FORM_ID) {
-      return await handleSuForm(
-        answers,
-        formId,
-        parsedBody.form_response.hidden.neighborhood,
-        parsedBody.form_response.hidden.broadcast_channel,
-      );
+    console.debug("[whebhook]", typeformType, JSON.stringify(answers));
+
+    if (isNotPartOfNeighborhood(answers)) {
+      return okResponse("user should live in neighborhood");
     }
 
-    if (formId === env.WAY_OF_LIFE_FORM_ID) {
-      return await handleWayOfLifeForm(
-        answers,
-        formId,
-        parsedBody.form_response.hidden?.neighborhood,
-        parsedBody.form_response.hidden?.broadcast_channel,
+    if (isUnder15(answers)) {
+      return okResponse("user should be under 15");
+    }
+
+    // could throw zod exception from zod parsing
+    const parsedAnswer = typeformSchemaMapper[typeformType].parse(answers) as
+      | ConvertedSuAnswer
+      | ConvertedWayOfLifeAnswer;
+
+    // remove isNeighborhoodResident property before save
+    // @ts-expect-error remove su name
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { isNeighborhoodResident, su, ...createQuery } = parsedAnswer;
+
+    const { surveyName, survey } = await getSurveyInformations(
+      parsedBody.form_response.hidden.neighborhood,
+      typeformType,
+    );
+
+    const validPhase = getValidSurveyPhase(typeformType);
+    if (isNotInPhase(survey, validPhase)) {
+      return notInPhaseSuSurveyResponse(surveyName, validPhase);
+    }
+
+    const broadcastChannel = parsedBody.form_response.hidden.broadcast_channel;
+
+    if (typeformType === TypeformType.SU) {
+      await createSu(
+        { ...createQuery, broadcastChannel } as SuAnswer,
+        surveyName,
+      );
+    } else {
+      let suId: number | undefined = undefined;
+      if ((parsedAnswer as ConvertedWayOfLifeAnswer).su) {
+        const su = await getOneSuBySu(
+          survey.id,
+          (parsedAnswer as ConvertedWayOfLifeAnswer).su!,
+        );
+        suId = su.id;
+      }
```

Comment:
:green_circle: J'ai eu un peu de mal à comprendre ce que cette partie faisait. Mes recos : 
- Le else c'est forcément un way of life donc autant faire un else if (typeformType === TypeformType.WayOfLife) plutot qu'un else. Et éventuellement rajouter un cas else (pas couvert actuellement ?)
- Mettre la logique dans une méthode du genre getSuIdFromSuNameOrThrow